posydon/binary_evol/DT/step_detached.py

```diff
@@ -1886,21 +1805,18 @@ def get_star_profile(star, htrack, m0):
                         binary.state = "RLO1"
                         binary.event = "oRLO1"
                 
-                if (binary.state == "RLO1" 
-                    and binary.star_1.state in LIST_ACCEPTABLE_STATES_FOR_HeStar 
-                    and binary.star_2.state in STAR_STATES_H_RICH):
+                if ('step_HMS_HMS_RLO' not in BINARY_STEPS_ALL
+                    and ((binary.star_1.state in STAR_STATES_HE_RICH_EVOLVABLE 
+                         and binary.star_2.state in STAR_STATES_H_RICH_EVOLVABLE)
+                    or (binary.star_1.state in STAR_STATES_H_RICH_EVOLVABLE
+                         and binary.star_2.state in STAR_STATES_HE_RICH_EVOLVABLE)
+                    or (binary.star_1.state in STAR_STATES_H_RICH_EVOLVABLE
+                         and binary.star_2.state in STAR_STATES_H_RICH_EVOLVABLE))):
```

Comment:
The `step_HMS_HMS_RLO` will only help for the case that both stars have a state in `STAR_STATES_H_RICH_EVOLVABLE`.
The cases where we have a `STAR_STATES_HE_RICH_EVOLVABLE` in there, the RLO grid of the HMS-HMS won't help either and the binaries still need to be send to failed.

Reply:
Hm, yes the flow although redirects the He starts to that step. But in the step itself they directly fail, because they are not in `FOR_RLO_STATES`. Again, we can may address this in a different PR, if so please open an issue.

Reply:
I've created Issue #448 for this.